src/data/onChainPoolDataViaReadContract.ts

```diff
@@ -18,24 +30,22 @@ type Result =
 
 type Results = Result[];
 
-const abi = parseAbi([
-    'function getPoolTokens(bytes32 poolId) view returns (address[] tokens, uint256 lastChangeBlock)',
-    'function getSwapFeePercentage() view returns (uint256)',
-    'function percentFee() view returns (uint256)',
-    'function protocolPercentFee() view returns (uint256)',
-    'function getNormalizedWeights() view returns (uint256[])',
-    'function totalSupply() view returns (uint256)',
-    'function getVirtualSupply() view returns (uint256)',
-    'function getActualSupply() view returns (uint256)',
-    'function getTargets() view returns (uint256 lowerTarget, uint256 upperTarget)',
-    'function getTokenRates() view returns (uint256, uint256)',
-    'function getWrappedTokenRate() view returns (uint256)',
-    'function getAmplificationParameter() view returns (uint256 value, bool isUpdating, uint256 precision)',
-    'function getPausedState() view returns (bool)',
-    'function inRecoveryMode() view returns (bool)',
-    'function getRate() view returns (uint256)',
-    'function getScalingFactors() view returns (uint256[] memory)',
-]);
+const abis = [
+    ...ComposableStablePoolV5Abi,
+    ...ConfigurableRightsPoolAbi,
+    ...ConvergentCurvePoolAbi,
+    ...FxPoolAbi,
+    ...GyroEV2Abi,
+    ...LiquidityBootstrappingPoolAbi,
+    ...ManagedPoolAbi,
+    ...MetaStablePoolAbi,
+    ...PhantomStablePoolAbi,
+    ...WeightedPoolAbi,
+    ...VaultAbi
```

Comment:
You should be able to reuse the existing vault abi if you move to .ts instead of .json abis.

Reply:
yes. removed the old one.